validator_client/src/block_service.rs

```diff
@@ -558,119 +536,154 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
                 RequireSynced::No,
                 OfflineOnFailure::Yes,
                 |beacon_node| async {
-                    self.publish_signed_block_contents::<Payload>(
-                        &signed_block_contents,
-                        beacon_node,
-                    )
-                    .await
+                    self.publish_signed_block_contents(&signed_block, beacon_node)
+                        .await
                 },
             )
             .await?;
 
         info!(
             log,
             "Successfully published block";
-            "block_type" => ?Payload::block_type(),
-            "deposits" => signed_block_contents.signed_block().message().body().deposits().len(),
-            "attestations" => signed_block_contents.signed_block().message().body().attestations().len(),
+            "builder_proposal" => builder_proposal,
```

Comment:
Would the type of `SignedBlock` be more useful here? The beacon node could still return a locally built payload even if `builder_proposal` is true (if BN is not connected to a builder, or if local block is more profitable)

Reply:
great point!